### PR TITLE
 Fix for Showcase overflow into secondary column

### PR DESCRIPTION
**Pull request by katebee**
#### 👉 Make Standard JS responsible for padding the secondary column

This PR adds a new function that uses fastdom to read the showcase image height, and then write the `padding-top` of the `js-secondary-column` to a value that will prevent the contents of the `js-secondary-column` from colliding with the Showcase image.

The ideal solution would be to change the html structure of showcase pages to reflect the design...

#### 👉 Restrict sizes of the aside slot on Showcase articles to the MPU sizes

When a DMPU loads into the RHS slot it can cause the most-viewed to be pushed down into the space occupied by inline2. Restricting the slot to a the shorter sizes will prevent this from occurring.

Changes have been made to the aside slot fragment and the support methods to support this on the serverside. 

## What is the value of this and can you measure success?

The right-hand-side (RHS) slot and the most viewed component both use the `js-secondary-column`. Adverts get a higher z-index than most over things on the page which is why it looks like the advert is invading the image space, when it is actually the other way around! 

This image overflow has mostly been okay due to a css workaround using something called `showcaseAdPadding`. However this relies on the image dimensions being close to 5:3 which is often not the case:

screen shot 2019-03-01 at 13 14 18

Removing the RHS slot from these pages would cause the most viewed component to be affected instead, hence the drastic action of adding JS to standard.

###### most-viewed colliding with the Showcase image when ads disabled:

image

###### RHS slot colliding with the Showcase image:

screen shot 2019-03-01 at 18 18 20

###### most-viewed colliding with inline2 when a DMPU loads into the RHS slot:

screen shot 2019-02-28 at 13 39 21

###### adverts looking good and no longer pushing most-viewed around: 

screen shot 2019-03-01 at 18 17 10

###### adfree looking good and no more hidden most-viewed:

screen shot 2019-03-01 at 18 14 17

### Does this affect other platforms?

Nope

### Does this affect GLabs Paid Content Pages?

Good question! Yes it does, and I just realised we could make a small improvement! This fix is still needed to prevent the adslot overlapping issue, however IIRC, GLabs pages do not include the most-viewed component, so we could allow DMPUs to be inserted.
 
- [ ] Check if we should allow DMPUs on Labs content

### Does this change break ad-free?

This change actually fixes Showcase articles for adfree users ✨ 🦄 🌈 

### Does this change update the version of CAPI we're using?

Nope

### Tested:

- [x] Locally
- [x] On CODE

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/article-aside-adverts.js [hunk 1]
```diff
@@ -18,22 +20,23 @@ const calculateAllowedAdSlots = (availableSpace: number): string => {
...
-        return Promise.resolve(false);
-    }
-
     const $mainCol: bonzo = $('.js-content-main-column');
     const $adSlot: bonzo = $('.js-ad-slot', $col);
     const $immersiveEls: bonzo = $('.element--immersive', $mainCol);
 
-    if (!$adSlot.length || !$mainCol.length) {
+    // article aside ads are added server-side UNLESS the page has a ShowcaseMainElement!
+    if (!$col.length || $col.css('display') === 'none') {
```
Comment (gtrufitt) [inline static/src/javascripts/projects/commercial/modules/article-aside-adverts.js]: Why was this moved below? the above `const` creations? I think you're using `$col` and doing a little bit where you don't need to. I'm assuming this is where some complexity has been remove, as you don't check for `adslot` or `maincol` anymore?

File: static/src/javascripts/projects/commercial/modules/article-aside-adverts.js [hunk 2]
```diff
@@ -46,18 +49,37 @@ export const init = (start: () => void, stop: () => void): Promise<boolean> => {
...
+
+                asideSlots.forEach(adSlot => {
+                    slotWrapper.append(adSlot);
+                });
+                return fastdom
+                    .write(() => {
+                        $col.prepend(slotWrapper);
+                    })
+                    .then(() => {
+                        addSlot(asideSlots[0], false);
```
Comment (gtrufitt) [inline static/src/javascripts/projects/commercial/modules/article-aside-adverts.js]: My fastdom knowledge is largely lost to the annals of time and I guess deep somewhere in `addSlot` maybe there's a write or read but I *think* maybe these could be in the write?

**Review by gtrufitt — APPROVED** [review]

File: static/src/javascripts/projects/commercial/modules/article-aside-adverts.js (hunk 1 above)
Comment (katebee) [reply static/src/javascripts/projects/commercial/modules/article-aside-adverts.js]: Good point! Think I was just hacking around with the code, will switch it back. 👍

File: static/src/javascripts/bootstraps/standard/main.js [hunk 3]
```diff
@@ -261,6 +262,11 @@ const bootStandard = (): void => {
 
     newHeaderInit();
 
+    if (config.get('page.hasShowcaseMainElement')) {
+        //
```
Comment (SiAdcock) [inline static/src/javascripts/bootstraps/standard/main.js]: It looks like you were about to say something! 👻

Comment (katebee) [reply static/src/javascripts/bootstraps/standard/main.js]: Yes! Then I realised that I could write some tests for the module, rather than comments here! 😜 👻 

Will remove this and add some comments and tests for fix-secondary-column 👍

File: static/src/javascripts/projects/commercial/modules/article-aside-adverts.js [hunk 4]
```diff
@@ -18,22 +20,23 @@ const calculateAllowedAdSlots = (availableSpace: number): string => {
     return '1,1|2,2';
 };
 
+const createSlotWrapper = (): Element => {
```
[the remark this answers is not in the document]
Comment (katebee) [reply static/src/javascripts/projects/commercial/modules/article-aside-adverts.js]: Good point! I have taken a second attempt at this and worked out how to create the appropriate adslot for showcase articles:

screen shot 2019-03-04 at 17 27 00

I have subsequently torn out a bunch of the JS 😄

**Review by jeteve — APPROVED** [review]

File: static/src/javascripts/projects/commercial/modules/article-aside-adverts.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (gtrufitt) [reply static/src/javascripts/projects/commercial/modules/article-aside-adverts.js]: 🥇
